### DISCUSSION TITLE
Have you considered providing a ThemeProvider?

### DISCUSSION THREAD

**Discussion by kai-orion** [material-components/material-web#5218] [Ideas]
Hi everyone, I would like to offer a suggestion.

I think you should provide a component called **ThemeProvider(e.g**. MaterialWebProvider), This component **will provide a set of color tokens and switch theme modes** through a few simple properties(sourceColor, dark).

Like other frameworks (such as fluent react), they all provide such a component, which can quickly create a set of tokens through a few simple parameters, which will greatly improve the efficiency of the code.

**Comment by asyncliz** [comment]
Yes! We've done some exploring with an `<md-theme>` component that does something like that.

We also would like to work on providing Sass and Typescript-based APIs to help generate and use tokens. It's on [our roadmap](https://github.com/material-components/material-web/blob/main/docs/roadmap.md#future) for the future.
